src/System.Diagnostics.DiagnosticSource/src/DiagnosticSourceUsersGuide.md

```diff
@@ -0,0 +1,377 @@
+	2. If you need to send more than one item, you can do so by creating a object with all information
+	   in it as properties.   CSharp's [anonymous type](https://msdn.microsoft.com/en-us/library/bb397696.aspx)
+	   feature is typically used to create a type to pass 'on the fly', and makes this scheme very
+	   convenient.     
+	3. At the instrumentation site, you must guard the call to 'Write' with an 'IsEnabled' check on 
+       the same event name.   Without this check, even when the instrumentation is inactive, the rules
+	   of the C# language require all the work of creating the payload object and calling 'Write' to be 
+	   done, even though nothing is actually listening for the data. By guarding the 'Write' call, we 
+	   make it efficient when the source is not enabled.   
+
+### Creating DiagnosticSources (Actually DiagnosticListeners)
+	   
+The code above assumes there is an DiagnosticSource called 'mySource'.   Perhaps confusingly you 
+make a DiagnosticSource by creating DiagnosticListener
+
+	static DiagnosticSource mySource = new DiagnosticListener("System.Net.Http");
+
+Basically a DiagnosticListener is a named place where a source sends its information (events).  
+From an implementation point of view, DiagnosticSource is a abstract class that has the two
+instrumentation methods, and DiagnosticListener is something that implements that abstract class.
+Thus every DiagnosticListener is a DiagnosticSource.   
+
+DiagnosticListener have a name, which is used to represent the component associated with the event.
+Thus the event names only need to be unique within a component.  
+
+----------------------------------------
+## Best Practices
+
+### Naming Conventions
+
+#### DiagnositicListener Names
+
+ * CONSIDER - the likely scenarios for USING information when deciding how may 
+   DiagnositicListener to have and the events in each.   Keep in mind that it is **very easy
+   and efficient** to filter all the events in a particular listener so ideally the 
+   most important scnearios involve turning in whole listners.    You may need to split
+   a source into multiple smaller ones to achieve this and this is OK.   For example There
+   are both incomming Http request and outgoing Http requests and you may only need one 
+   or the other, so having a System.Net.Http.Incomming and System.Net.Http.OutGoing for
+   each sub-case is good.  
+
+ * CONSIDER -  the likely volume of events.   High volume events may deserve their own
+   DiagnositicListener.   You don't really want to mix high volume and low volume events
+   in the same listener unless they both support the same scenario.   It is OK however to
+   put several **low volume** events in a 'miscelaneous' listener, even if they support different 
+   scenarios if it simplifies things enough.
+
+ * DO - Consider the scenario when picking the name for the DiagnositicListener.  Often This
```

Comment:
Nit: DiagnositicListener => DiagnosticListener
Nit: "Often This" => "Often, this"